AVIenabledbyArc/vi-edge-deployment-script.sh

```diff
@@ -37,126 +27,122 @@ ask_question() {
         eval "$variable=\"$input\""
     fi
 }
-##############################
-# create_cognitive_hobo_resources
-# Creeating Cognitive Services On VI RP, on behalf of the user
-##############################
+
+##############################################
+#  CLI Pre-requisites
+###############################################
+function install_cli_tools {
+  # https://learn.microsoft.com/en-us/azure/azure-arc/kubernetes/quickstart-connect-cluster?tabs=azure-cli
+  echo "ensure you got the latest CLI client and install add ons if needed"
+  echo "https://learn.microsoft.com/en-us/azure/azure-arc/kubernetes/quickstart-connect-cluster?tabs=azure-cli"
+  az extension add --name connectedk8s
+  az extension add --name aks-preview
+  az provider register --namespace Microsoft.Kubernetes
+  az provider register --namespace Microsoft.KubernetesConfiguration
+  az provider register --namespace Microsoft.ExtendedLocation
+}
+
+##################################################################
+# Create Cognitive Services HOBO (On Behalf Of the User ) Resources
+##################################################################
 function create_cognitive_hobo_resources {
   echo -e "\t create Cognitive Services On VI RP ***start***"
-  sleepDuration=90
+  sleepDuration=10
   echo "getting arm token"
-  createResourceUri="https://management.azure.com/subscriptions/${viSubscriptionId}/resourceGroups/${viResourceGroup}/providers/Microsoft.VideoIndexer/accounts/${viAccountName}/CreateExtensionDependencies?api-version=2023-06-02-preview"
+  createResourceUri="https://management.azure.com/subscriptions/${subscriptionId}/resourceGroups/${resourceGroup}/providers/Microsoft.VideoIndexer/accounts/${accountName}/CreateExtensionDependencies?api-version=${viApiVersion}"
   echo "=============================="
   echo "Creating cs resources"
   echo "=============================="
   result=$(az rest --method post --uri $createResourceUri 2>&1 >/dev/null || true)
   echo $result    
 
-  if [[ "$result" == *"ERROR:"* ]]; then
-    echo "CS Resources already exist. ignoring"
+  if [[ "$result" == *"ERROR: Conflict"* ]]; then
+    echo "CS Resources already exist. Moving on."
   else
-    echo "sleeping for $sleepDuration seconds"  
-    sleep $sleepDuration
+    echo "No CS resources found. Creating"  
```

Comment:
This is a bug, we should check that the response is 202. that is the only case where we create.
If we get an error that is not 202 or conflict, we should fail the script

Reply:
if we are creating in the background new resource , we need to wait for them to provision , no ?